### PR TITLE
(PUP-10260) Add rest terminus tests

### PR DESCRIPTION
**Pull request by joshcooper**
Adds tests for rest termini. Use webmock instead of manually stubbing Net::HTTP.

### CODE REVIEW

File: spec/unit/indirector/report/rest_spec.rb [hunk 1]
```diff
@@ -22,71 +35,62 @@
...
+    expect(described_class.indirection.save(report)).to be_nil
   end
 
-  describe "#save" do
-    let(:response) { mock_response(200, 'body') }
-    let(:connection) { double('mock http connection', :put => response, :verify_callback= => nil) }
-    let(:instance) { model.new('the thing', 'some contents') }
-    let(:request) { save_request(instance.name, instance) }
-    let(:body) { ["store", "http"].to_pson }
+  it "raises if fail_on_404 is specified" do
```
Comment (joshcooper) [inline spec/unit/indirector/report/rest_spec.rb]: test is empty

Comment (joshcooper) [reply spec/unit/indirector/report/rest_spec.rb]: fail_on_404 is not valid for put requests (they don't take options)

File: spec/unit/indirector/file_metadata/rest_spec.rb [hunk 2]
```diff
@@ -3,8 +3,115 @@
...
+    it "returns an empty array if the metadata doesn't exist" do
+      stub_request(:get, uri).to_return(status: 404)
+
+      expect(described_class.indirection.search(key)).to eq([])
+    end
+
+    it "returns an empty array if the metadata doesn't exist and fail_on_404 is true" do
+      stub_request(:get, uri).to_return(status: 404, headers: { 'Content-Type' => 'application/json'}, body: "{}")
+
+      expect(described_class.indirection.search(key, fail_on_404: true)).to eq([])
```
Comment (melissa) [inline spec/unit/indirector/file_metadata/rest_spec.rb]: In the previous PR, I commented on this, and you responded, but I'm still not sure I follow 100%
>if fail_on_404 is false, then the old implementation returns an empty array
Isn't that the opposite of what's happening here? Or was that a typo? Should we add another test case where `fail_on_404: false`? That would be the same as the test on line 88, right? In which case, both scenarios do the same thing?

Or is this test just establishing that `fail_on_404: true` doesn't impact what gets returned if the metadata doesn't exist?

Comment (joshcooper) [reply spec/unit/indirector/file_metadata/rest_spec.rb]: Sorry I may have conflated find and search in my answer. For this PR, HTTP 404 causes search to always return with an empty array regardless of `fail_on_404`. For find, we either raise (if fail_on_404 is true) or we return an empty array (if fail_on_404 is nil/false).

For this PR, yes `fail_on_404` only affects find. For search, we always return an empty array if fail_on_404 is true or false.

Comment (melissa) [reply spec/unit/indirector/file_metadata/rest_spec.rb]: Cool, thank you for that clarification

**Pull request merged**